core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolution.js

```diff
@@ -0,0 +1,277 @@
+              }
+
+              $modalInstance.close({
+                solution: angular.copy(
+                  SolutionObjectFactory.createNew(
+                    $scope.tmpSolution.answerIsExclusive,
+                    answer,
+                    $scope.tmpSolution.explanation))
+              });
+            };
+
+            $scope.cancel = function() {
+              $modalInstance.dismiss('cancel');
+              alertsService.clearWarnings();
+            };
+          }
+        ]
+      }).result.then(function(result) {
+        var explorationId =
+          explorationContextService.getExplorationId();
+        var currentStateName =
+          editorContextService.getActiveStateName();
+        var state = explorationStatesService.getState(currentStateName);
+        var interactionId = stateInteractionIdService.savedMemento;
+        var rulesServiceName = (
+          angularNameService.getNameOfInteractionRulesService(interactionId));
+        // Inject RulesService dynamically.
+        var rulesService = $injector.get(rulesServiceName);
+        var correctAnswer = result.solution.correctAnswer;
+        try {
+          AnswerClassificationService.getMatchingClassificationResult(
+            explorationId, state, correctAnswer, true, rulesService)
+            .then(function (result) {
+              if (editorContextService.getActiveStateName() !== (
+              result.outcome.dest)) {
+                stateSolutionService.saveDisplayedValue();
+                $scope.solutionIsSpecified = (
+                  !!JSON.stringify(stateSolutionService.savedMemento));
+              } else {
+                alertsService.addInfoMessage('That solution does not lead ' +
+                  'to the next state!');
+                $scope.openAddSolutionModal();
+              }
+            });
+        } catch (e) {
+          alertsService.addInfoMessage('That solution was invalid!');
+          $scope.openAddSolutionModal();
+        }
+        stateSolutionService.displayed = result.solution;
+      });
+    };
+
+    $scope.deleteSolution = function(evt) {
+      // Prevent clicking on the delete button from also toggling the display
+      // state of the answer group.
+      evt.stopPropagation();
+
+      alertsService.clearWarnings();
+      $modal.open({
+        templateUrl: 'modals/deleteSolution',
+        backdrop: true,
+        scope: $scope,
+        controller: [
+          '$scope', '$modalInstance', function($scope, $modalInstance) {
+            $scope.reallyDelete = function() {
+              $modalInstance.close();
+            };
+
+            $scope.cancel = function() {
+              $modalInstance.dismiss('cancel');
+              alertsService.clearWarnings();
+            };
+          }
+        ]
+      }).result.then(function() {
+        stateSolutionService.displayed = null;
+        stateSolutionService.saveDisplayedValue();
+        $scope.solutionIsSpecified = false;
```

Comment:
Incidentally, I'm starting to question why you even need this variable at all. Can't you just do Boolean(solution) or something like that where it's needed? Otherwise you basically have two different sources of truth for the solution's existence.